nip19.ts

```diff
@@ -39,6 +39,14 @@ export type DecodeResult =
   | {type: 'npub'; data: string}
   | {type: 'note'; data: string}
 
+export function decode(nip19: `nprofile1${string}`): {type: 'nprofile'; data: ProfilePointer}
+export function decode(nip19: `nrelay1${string}`): {type: 'nrelay'; data: string}
+export function decode(nip19: `nevent1${string}`): {type: 'nevent'; data: EventPointer}
+export function decode(nip19: `naddr1${string}`): {type: 'naddr'; data: AddressPointer}
+export function decode(nip19: `nsec1${string}`): {type: 'nsec'; data: string}
+export function decode(nip19: `npub1${string}`): {type: 'npub'; data: string}
+export function decode(nip19: `note1${string}`): {type: 'note'; data: string}
+export function decode(nip19: string): DecodeResult
```

Comment:
There might be some way to reduce boilerplate/duplication with the code above, but I'm not sure how yet. We could improve it in a future MR.

Reply:
Figured out how to make it DRY. Thanks ChatGPT. https://chat.openai.com/share/f09f38e5-f541-4f98-9483-e183f5650398